boards/arm/bbc_microbit/bbc_microbit.dts

```diff
@@ -40,6 +41,46 @@
 		};
 	};
 
+	led_matrix {
+		compatible = "gpio-leds";
+		row_1: led_row1 {
+			gpios = <&gpio0 13 GPIO_ACTIVE_HIGH>;
+		};
+		row_2: led_row2 {
+			gpios = <&gpio0 14 GPIO_ACTIVE_HIGH>;
+		};
+		row_3: led_row3 {
+			gpios = <&gpio0 15 GPIO_ACTIVE_HIGH>;
+		};
+
+		col_1: led_col1 {
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+		};
+		col_2: led_col2 {
+			gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
```

Comment:
suggestion: create a specific compatible and create something like:

```
led_matrix {
    compatible = "bbc,microbit-matrixled";
    row-gpios = <&gpio0 13 GPIO_ACTIVE_HIGH>, <&gpio0 14 GPIO_ACTIVE_HIGH>, ...;
    col-gpios = <&gpio0 4 GPIO_ACTIVE_LOW>, ...;
```

Reply:
I'm trying to follow the approach. But how then use these row_gpios and col_gpios in a universal way for anything? For MicrobitV1 row-gpios contains 3 elements and col-gpios 9 elements. For MicrobitV2 they are 5 and 5 elements. Ok, with `DT_INST_PROP_LEN(0, row_gpios)` it is possible to know how many elements are in these `phandle-arrays`. But how then this information can be used for pin initialization and for pin settings? 
```
for(uint8_t i=0; i < DT_INST_PROP_LEN(0, row_gpios); i++) {
     GPIO_DT_SPEC_INST_GET_BY_IDX(0, row_gpios,i)
}
``` 

doesn't work.  
Is there any way to iterate over all `phandle-array` elements to generate an initialization structure like
```
static const struct gpio_dt_spec row_gpio_dt_spec[] = {
   {GPIO_DT_SPEC_INST_GET_BY_IDX(0, row-gpios,0)},
   {GPIO_DT_SPEC_INST_GET_BY_IDX(0, row-gpios,1)},
   ...
   {GPIO_DT_SPEC_INST_GET_BY_IDX(0, row-gpios,DT_INST_PROP_LEN(0, row_gpios))},
}; 
```
?

I thought `DT_INST_FOREACH_CHILD` does the magic, but I failed with it.

Would you please help?

Reply:
I can answer to the questions by myself: it is needed to create configuration for the driver manually. This is a board-specific static array that actually maps logical row and column numbers to these DT elements. The initialization then is done by iteration over elements of this array (actually, arrays).

Reply:
Magic fixed, proposal implemented.